beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -270,23 +281,22 @@ func (vs *Server) getPayloadHeaderFromBuilder(
 		return nil, errors.Wrap(err, "could not validate builder signature")
 	}
 
-	maxBlobsPerBlock := params.BeaconConfig().MaxBlobsPerBlock(slot)
 	var kzgCommitments [][]byte
 	if bid.Version() >= version.Deneb {
-		kzgCommitments, err = bid.BlobKzgCommitments()
-		if err != nil {
-			return nil, errors.Wrap(err, "could not get blob kzg commitments")
+		dBid, ok := bid.(builder.BidDeneb)
+		if !ok {
+			return nil, errors.New("builder returned non-deneb or above bid")
 		}
-		if len(kzgCommitments) > maxBlobsPerBlock {
-			return nil, fmt.Errorf("builder returned too many kzg commitments: %d", len(kzgCommitments))
-		}
-		for _, c := range kzgCommitments {
-			if len(c) != fieldparams.BLSPubkeyLength {
-				return nil, fmt.Errorf("builder returned invalid kzg commitment length: %d", len(c))
-			}
+		kzgCommitments = dBid.BlobKzgCommitments()
+	}
+	var executionRequests *enginev1.ExecutionRequests
+	if bid.Version() >= version.Electra {
+		eBid, ok := bid.(builder.BidElectra)
+		if !ok {
+			return nil, errors.New("builder returned non-electra or above bid")
```

Comment:
```suggestion
			return fmt.Errorf("bid type %T does not implement builder.BidElectra", eBid)
```
For unity with above warning logs